scripts/Muon/GUI/Common/fitting_widgets/tf_asymmetry_fitting/tf_asymmetry_fitting_model.py

```diff
@@ -0,0 +1,964 @@
+
+        dataset_name = parameters["ReNormalizedWorkspaceList"]
+        CopyLogs(InputWorkspace=dataset_name, OutputWorkspace=output_workspace, StoreInADS=False)
+        self._add_single_fit_results_to_ADS_and_context(dataset_name, parameter_table, output_workspace,
+                                                        covariance_matrix)
+        return function, fit_status, chi_squared
+
+    def _do_tf_asymmetry_simultaneous_fit(self, parameters: dict, global_parameters: list) -> tuple:
+        """Performs a TF Asymmetry fit in simultaneous fit mode."""
+        output_workspace, parameter_table, function, fit_status, chi_squared, covariance_matrix = \
+            self._run_tf_asymmetry_fit(parameters)
+
+        dataset_names = parameters["ReNormalizedWorkspaceList"]
+        self._copy_logs(dataset_names, output_workspace)
+        self._add_simultaneous_fit_results_to_ADS_and_context(dataset_names, parameter_table, output_workspace,
+                                                              covariance_matrix, global_parameters)
+        return function, fit_status, chi_squared
+
+    def _run_tf_asymmetry_fit(self, parameters: dict) -> tuple:
+        """Performs the TF Asymmetry fit using the parameters dict provided."""
+        alg = AlgorithmManager.create("CalculateMuonAsymmetry")
+        return run_CalculateMuonAsymmetry(parameters, alg)
+
+    def _get_parameters_for_tf_asymmetry_single_fit(self, dataset_name: str, tf_single_function: IFunction) -> dict:
+        """Returns the parameters to use for a single TF Asymmetry fit."""
+        params = self._get_common_tf_asymmetry_parameters()
+        params["InputFunction"] = tf_single_function.clone()
+        params["ReNormalizedWorkspaceList"] = dataset_name
+        params["UnNormalizedWorkspaceList"] = self._get_unnormalised_workspace_list([dataset_name])[0]
+
+        fit_workspace_name, _ = create_fitted_workspace_name(dataset_name, self.function_name)
+        params["OutputFitWorkspace"] = fit_workspace_name
+        return params
+
+    def _get_parameters_for_tf_asymmetry_simultaneous_fit(self, dataset_names: list,
+                                                          tf_simultaneous_function: IFunction) -> dict:
+        """Returns the parameters to use for a simultaneous TF Asymmetry fit."""
+        params = self._get_common_tf_asymmetry_parameters()
+        params["InputFunction"] = str(tf_simultaneous_function) + self._construct_global_tie_appendage()
+        params["ReNormalizedWorkspaceList"] = dataset_names
+        params["UnNormalizedWorkspaceList"] = self._get_unnormalised_workspace_list(dataset_names)
+
+        fit_workspace_name, _ = create_multi_domain_fitted_workspace_name(dataset_names[0], self.function_name)
+        params["OutputFitWorkspace"] = fit_workspace_name
+        return params
+
+    def _get_common_tf_asymmetry_parameters(self) -> dict:
+        """Returns the common parameters used for a TF Asymmetry fit."""
```

Comment:
Are these not just common parameters for any type of fit? + Everything in these methods is just dependent on the base class, so maybe it could be implemented there?

Reply:
The TF Asymmetry mode doesn't use the `SecondPulseWeight` parameter for double pulse fitting, so there is a slight difference. The `CalculateMuonAsymmetry` algorithm (used for TF fitting) also doesnt seem to have an `EvaluationType` property, so these slight differences meant I had to do this differently.